src/OrchardCore.Modules/OrchardCore.Media.Azure/Services/MediaBlobContainerTenantEvents.cs

```diff
@@ -56,28 +56,65 @@ public override async Task ActivatingAsync()
 
     public override async Task RemovingAsync(ShellRemovingContext context)
     {
-        // Only remove container if options are valid.
-        if (!_options.RemoveContainer || !_options.IsConfigured())
+        // Only carry out removal if options are valid.
+        if (!_options.IsConfigured() || (!_options.RemoveFilesFromBasePath && !_options.RemoveContainer))
         {
             return;
         }
 
-        try
+        var blobContainer = new BlobContainerClient(_options.ConnectionString, _options.ContainerName);
+        if (_options.RemoveContainer)
         {
-            var blobContainer = new BlobContainerClient(_options.ConnectionString, _options.ContainerName);
+            try
+            {
+                var response = await blobContainer.DeleteIfExistsAsync();
+                if (!response.Value)
+                {
+                    _logger.LogError("Unable to remove the Azure Media Storage Container {ContainerName}.", _options.ContainerName);
 
-            var response = await blobContainer.DeleteIfExistsAsync();
-            if (!response.Value)
+                    context.ErrorMessage = S["Unable to remove the Azure Media Storage Container '{0}'.", _options.ContainerName];
+                }
+            }
+            catch (RequestFailedException ex)
             {
-                _logger.LogError("Unable to remove the Azure Media Storage Container {ContainerName}.", _options.ContainerName);
-                context.ErrorMessage = S["Unable to remove the Azure Media Storage Container '{0}'.", _options.ContainerName];
+                _logger.LogError(ex, "Failed to remove the Azure Media Storage Container {ContainerName}.", _options.ContainerName);
+
+                context.ErrorMessage = S["Failed to remove the Azure Media Storage Container '{0}'.", _options.ContainerName];
+                context.Error = ex;
+            }
+
+            // If both deletion options are set, return here to avoid errors when trying to delete files from a non-existent container.
+            if (_options.RemoveFilesFromBasePath)
```

Comment:
You can just return here, right?

Reply:
Think you're right, yeah, the additional check isn't necessary.